### PR TITLE
Assignment 6

### PR DESCRIPTION
**Pull request by prassr**
Created assignment 6.

@hashkrish

### CODE REVIEW

File: assignment_6/evaluation_file.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+for i in {1..10}; do
+    c=`shuf -i 0-$(( $num_cities-1 )) -n 1`
+    idx2=`shuf -i 0-1 -n 1`
+    if [ ${idx2} -eq 0 ]; then
+       idx3=`shuf -i 0-1 -n 1`
+       a3=${arg3[idx3]}
+       #echo $a3
+    else
+       idx3=`shuf -i 2-3 -n 1`
+       if [ $idx3 -eq 2 ];
```
Comment (hashkrish) [inline assignment_6/evaluation_file.sh:66]:
```suggestion
       if [ "$idx3" -eq 2 ];
```

File: assignment_6/evaluation_file.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+       then
+           a3="mode";
+       else
+           a3="average";
+       fi
+    fi
+    ans1=${hash[${city[c]},${idx3}]}
+    ans2=$( ./script.sh ${city[c]} ${arg2[${idx2}]} $a3 || echo -1 ) 
+
+    if [ `echo "${ans1} == ${ans2}" | bc` -eq 0 ]; then
```
Comment (hashkrish) [inline assignment_6/evaluation_file.sh:76]: We can use `!=` for POSIX string inequality.
```suggestion
    if [  "${ans1}" != "${ans2}" ]; then
```

**Comment by hashkrish** [comment]
@prassr I have added some comments. Please check.

**Review by hashkrish — CHANGES_REQUESTED** [review]

File: assignment_6/evaluation_file.sh (hunk 2 above)
Comment (prassr) [reply assignment_6/evaluation_file.sh:73]: When the student returns 22.00 and the actual answer is 22, `bc` will return `true`, but the ` != ` will return `false`

Comment (hashkrish) [reply assignment_6/evaluation_file.sh:73]: Got it.

**Comment by hashkrish** [comment]
@prassr Please do a local testing with the latest pull. Let me know if everything is fine. I will merge it.

**Review by hashkrish — APPROVED** [review]

**Comment by prassr** [comment]
@hashkrish 
Checked locally, working fine , sir.

**Comment by hashkrish** [comment]
@prassr  Merged and deployed in the VM

**Pull request merged**